Tools/BlockTypePaletteGenerator/lib/init.lua

```diff
@@ -0,0 +1,3 @@
+-- lib/ is not in default Lua package paths
+package.path = './lib/?.lua;' .. package.path;
+package.cpath = './lib/?.so;' .. package.cpath;
```

Comment:
I'd prefer if the script could be run as a single command in the console (or from an IDE). For that, it needs to be self-contained.

Reply:
It's a single command: `lua ./Generator.lua`

I can add a shebang line and make it executable, so you can run just run `./Generator.lua`, if you prefer that

Reply:
Does lua load `init.lua` files automatically? I didn't know. And I'm not sure if my IDE does that (highly doubtful) (I'm using the excellent ZeroBrane Studio). Mainly for the IDE reasons, I'd prefer a script that takes parameters specifying the input file and output file name, rather than operating on stdin / stdout.

Reply:
`./?/init.lua` is in `package.path` by default, so `require "lib"` loads `./lib/init.lua`

Reply:
```
if (args[1] == "-") then
  fIn = io.input()
else
  fIn = assert(io.open(args[1], "r"), "Invalid input file: " .. args[1])
end
```
What library? :)